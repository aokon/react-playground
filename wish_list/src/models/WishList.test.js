import { WishListItem, WishList } from './WishList'
import { getSnapshot } from 'mobx-state-tree'

it('can create an instance of the WishListItem model', () => {
  const item = WishListItem.create({
    name: 'Test',
    price: 34.99,
  })

  expect(item.name).toBe('Test')
  expect(item.image).toBe('')

  item.changeName('New name')

  expect(item.name).toBe('New name')
})

it('can create WishList model', () => {
  const list = WishList.create({
    items: [{
      name: 'Test',
      price: 34.99,
    }]
  })

  expect(list.items.length).toBe(1)
})

it('can add a new WishListItem to the list', () => {
  const list = WishList.create()

  list.add(WishListItem.create({
    name: 'Dummy',
    price: 10
  }))

  expect(list.items.length).toBe(1)
  expect(getSnapshot(list)).toMatchSnapshot()
})
