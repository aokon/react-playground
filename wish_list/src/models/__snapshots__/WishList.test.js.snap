// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can add a new WishListItem to the list 1`] = `
Object {
  "items": Array [
    Object {
      "image": "",
      "name": "Dummy",
      "price": 10,
    },
  ],
}
`;
